pkg/api/peer_req_resp.go

```diff
@@ -18,6 +18,11 @@ type PeerAddReq struct {
 	MetaData  map[string]string `json:"metadata"`
 }
 
+type MetaDataEdit struct {
+	PeerID   uuid.UUID         `json:"peerid"`
```

Comment:
The peer id is an existing resource. This shall be part of the URL, not in the body.

I also see an opportunity to allow editing of any field of Peer object (such as name and addresses), not just metadata. There's a [related issue](https://github.com/gluster/glusterd2/issues/356)